backend/src/crd/controller/scheduledworkflow/controller.go

```diff
@@ -414,19 +422,19 @@ func (c *Controller) syncHandler(ctx context.Context, key string) (
 			wraperror.Wrapf(err, "Syncing ScheduledWorkflow (%v): transient failure, can't fetch completed workflows: %v", name, err)
 	}
 
-	workflow, nextScheduledEpoch, err := c.submitNextWorkflowIfNeeded(ctx, swf, len(active), nowEpoch)
+	submitted, nextScheduledEpoch, err := c.submitNextWorkflowIfNeeded(ctx, swf, len(active), nowEpoch)
```

Comment:
example change that can be extracted out

Reply:
> I'm not sure how big an ask this is, but I can see that we will need to discuss the design with v2 in context, so there might be more design discussions.
> 

As long as it bring us forward, I am fine with it. I will leave this one up for now but start to bring things out. Will start to take a look at your suggestion of the `submitNextWorkflowIfNeeded `.

Reply:
@Bobgy : https://github.com/kubeflow/pipelines/pull/6918